test/unit/search/kmer_index/shape_iterator_tests.cpp

```diff
@@ -0,0 +1,124 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include "gtest/gtest.h" // für googletest
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/search/kmer_index/shape.hpp>
+#include <seqan3/search/kmer_index/shape_iterator.hpp>
+#include <seqan3/range/view/take.hpp>
+
+using namespace seqan3;
+
+// ============================================================================
+//  shape
+// ============================================================================
+
+TEST(ShapeTest, HandlesSizes)
+{
+    EXPECT_EQ(std::ranges::size(shape()),    0u);
+    EXPECT_EQ(std::ranges::size(shape(0)),   0u);
+    EXPECT_EQ(std::ranges::size(shape(1)),   1u);
+    EXPECT_EQ(std::ranges::size(shape(31)), 31u);
+    EXPECT_EQ(std::ranges::size(shape(0,0)), 2u);
+    EXPECT_EQ(std::ranges::size(shape(1,1)), 2u);
+    EXPECT_EQ(std::ranges::size(shape(0,0,0,0,0,0)), 6u);
+    EXPECT_EQ(std::ranges::size(shape(1,0,1,0,1,1)), 6u);
+    EXPECT_EQ(std::ranges::size(shape(1,1,1,1,1,1)), 6u);
+}
+
+// size 3 ausreichend?
+TEST(ShapeTest, HandlesShapes)
+{
+    shape s0(0,0,0);
+    std::array<bool,3> a0{false,false,false};
+    for (size_t i(0); i < a0.size(); i++)
```

Comment:
`EXPECT_TRUE(std::ranges::equal(s0.get(), a0));`

Reply:
`EXPECT_TRUE(std::ranges::equal(s0.get(), a0));`
returns false when it should be true.

Reply:
```cpp
TEST(ShapeTest, HandlesShapes)
{
    shape s0(0, 0, 0);
    std::array<bool, 32> a0{false, false, false};
    EXPECT_EQ(s0.get(), a0);

    shape s1(1, 0, 1);
    std::array<bool, 32> a1{true, false, true};
    EXPECT_EQ(s1.get(), a1);

    shape s2(3);
    std::array<bool, 32> a2;
    a2.fill(true);
    EXPECT_EQ(s2.get(), a2);
}
```

Reply:
so this doesn't work because the strong types get()-method returns an array of type <type, 32>. They can never be equal as long as I check against an array with length 3.
Because of that, I'd suggest to leave it as it is.

Reply:
That's why I wrote  `std::array<bool, 32> a0{false, false, false};` with a 32 and not a 3 😁

Reply:
Ok, that's convenient :)